pkg/xds/server/callbacks/workload_label_validator_test.go

```diff
@@ -0,0 +1,625 @@
+package callbacks_test
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	common_api "github.com/kumahq/kuma/v2/api/common/v1alpha1"
+	mesh_proto "github.com/kumahq/kuma/v2/api/mesh/v1alpha1"
+	core_mesh "github.com/kumahq/kuma/v2/pkg/core/resources/apis/mesh"
+	meshidentity_api "github.com/kumahq/kuma/v2/pkg/core/resources/apis/meshidentity/api/v1alpha1"
+	core_manager "github.com/kumahq/kuma/v2/pkg/core/resources/manager"
+	core_model "github.com/kumahq/kuma/v2/pkg/core/resources/model"
+	core_store "github.com/kumahq/kuma/v2/pkg/core/resources/store"
+	core_xds "github.com/kumahq/kuma/v2/pkg/core/xds"
+	"github.com/kumahq/kuma/v2/pkg/plugins/resources/memory"
+	"github.com/kumahq/kuma/v2/pkg/plugins/runtime/k8s/metadata"
+	test_model "github.com/kumahq/kuma/v2/pkg/test/resources/model"
+	"github.com/kumahq/kuma/v2/pkg/util/pointer"
+	. "github.com/kumahq/kuma/v2/pkg/xds/server/callbacks"
+)
+
+var _ = Describe("Workload Label Validator", func() {
+	var resManager core_manager.ResourceManager
+	var validator *WorkloadLabelValidator
+
+	BeforeEach(func() {
+		memStore := memory.NewStore()
+		resManager = core_manager.NewResourceManager(memStore)
+		validator = NewWorkloadLabelValidator(resManager)
+
+		// Create default mesh
+		err := resManager.Create(context.Background(), core_mesh.NewMeshResource(), core_store.CreateByKey(core_model.DefaultMesh, core_model.NoMesh))
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("when MeshIdentity uses workload label in path template", func() {
+		BeforeEach(func() {
+			// Create MeshIdentity with workload label in path template
+			mi := &meshidentity_api.MeshIdentityResource{
+				Meta: &test_model.ResourceMeta{
+					Name: "mi-with-workload-label",
+					Mesh: "default",
+				},
+				Spec: &meshidentity_api.MeshIdentity{
+					Selector: &meshidentity_api.Selector{
+						Dataplane: &common_api.LabelSelector{
+							MatchLabels: &map[string]string{
+								"kuma.io/service": "web",
+							},
+						},
+					},
+					SpiffeID: &meshidentity_api.SpiffeID{
+						TrustDomain: pointer.To("{{ label \"kuma.io/mesh\" }}.mesh.local"),
```

Comment:
nitpick: you can use backtics ` to not have to escape strings like that

for example:
```suggestion
						TrustDomain: pointer.To(`{{ label "kuma.io/mesh" }}.mesh.local`),
```